src/Compilers/VisualBasic/Portable/VisualBasicParseOptions.vb

```diff
@@ -318,6 +239,32 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Get
         End Property
 
+
+        Friend Overrides Sub ValidateOptions(builder As ArrayBuilder(Of Diagnostic))
+            ValidateOptions(builder, MessageProvider.Instance)
+
+            ' Validate LanguageVersion Not SpecifiedLanguageVersion, after Latest/Default has been converted
+            If Not LanguageVersion.IsValid Then
+                builder.Add(Diagnostic.Create(MessageProvider.Instance, ERRID.ERR_BadLanguageVersion, LanguageVersion.ToString))
+            End If
+
+            If Not PreprocessorSymbols.IsDefaultOrEmpty Then
+                For Each symbol In PreprocessorSymbols
+                    If Not IsValidIdentifier(symbol.Key) OrElse SyntaxFacts.GetKeywordKind(symbol.Key) <> SyntaxKind.None Then
+                        builder.Add(Diagnostic.Create(MessageProvider.Instance, ERRID.ERR_ProjectCCError1, VBResources.ERR_ExpectedIdentifier, symbol.Key))
```

Comment:
Following up, there should be a different error message as these options are file-level, not project-level.

---
In reply to: [105271724](https://github.com/dotnet/roslyn/pull/17113#discussion_r105271724) [](ancestors = 105271724,101844385)